pkg/virtctl/adm/logverbosity/logverbosity.go

```diff
@@ -360,20 +382,109 @@ func findOperation(cmd *cobra.Command) (operation, error) {
 	}
 }
 
+func getVmiVerbosity(currentLv map[string]uint, vm *v1.VirtualMachine, vmNamespace string) error {
+	if *vm.Spec.Running {
+		// check pod object's environmental variable VIRT_LAUNCHER_LOG_VERBOSITY
+		podList, err := virtClient.CoreV1().Pods(vmNamespace).List(context.Background(), k8smetav1.ListOptions{})
```

Comment:
You could simplify this a little bit and use [labels on the pod](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/services/template.go#L1525-L1526) to filter the right pod

Reply:
Thank you for letting me know. That is great to know. I also want to find a way to simplify the nesting. However, I think this is not enough.

First, in the `*vm.Spec.Running` clause of `getVmiVerbosity()`, in order to show the verbosity of the vmi, we search the `VIRT_ LAUNCHER_LOG_VERBOSITY` environment variable in the `compute` container of the `virt-launcher` Pod corresponding to the vmi object. With the information the program has, there is no way to directly specify the `compute` container without the search.

The verbosity level of` virt-launcher` is set by an environment variable `VIRT_LAUNCHER_LOG_VERBOSITY` when the Pod is created. The environment variable is the value of `vmi.Labels[logVerbosity]`, otherwise the cluster-wide `virt-launcher` log verbosity value is used. If no `virt-launcher` log verbosity is specified, the environment variable is not set and verbosity is the default (2).
https://github.com/kubevirt/kubevirt/blob/v1.1.1/pkg/virt-controller/services/template.go#L420-L436

By the method you taught, when a Pod is created, the `Labels` set in the vmi object are copied to the Pod's `Labels`.
https://github.com/kubevirt/kubevirt/blob/v1.1.1/pkg/virt-controller/services/template.go#L539

So, if the `Labels` of the VMI object are set, the method will help. However, if the `Labels` of the VMI object are not set, the log verbosity of `virt-launcher` is used, but this log verbosity value is not reflected in the `Labels` of the Pod. So, this method does not help.

As mentioned above, to check the verbosity of a running VM, we need to refer to the `VIRT_LAUNCHER_LOG_VERBOSITY` environment variable that was used when the pod was started, not the `Labels` of Pod. So, I do not think we can simplify the code even if we use the method.

This is my understanding, but if I miss something, please let me know.

Reply:
@hshitomi my comment was more related to filter the virt-launher pods among all the pods in the namesoace not the verbosity

Reply:
Thanks for the clarification. Understand it. And, I fix it as the following.
```
labelSelector := labels.Set{
	v1.VirtualMachineNameLabel: vmiName,
}.AsSelector()
pods, err := virtClient.CoreV1().Pods(vmNamespace).List(context.Background(), k8smetav1.ListOptions{
	LabelSelector: labelSelector.String(),
})
if err != nil {
	return err
}
if len(pods.Items) == 0 {
	return fmt.Errorf("No pods found for the launcher")
}
for _, container := range pods.Items[0].Spec.Containers {
	if container.Name != "compute" {
		continue
	}
	for _, env := range container.Env {
		if env.Name == "VIRT_LAUNCHER_LOG_VERBOSITY" {
			currentLv["vmi"], err = atou(env.Value)
			if err != nil {
				return err
			}
			break
		}
		// if there is no VIRT_LAUNCHER_LOG_VERBOSITY, the virt-launcher verbosity was set as default
	}
}
```

Reply:
The code about the label selector works fine with my local cluster. However, the fake client does not work as I expect with the label selector. So the original code still remains.